src/status_im2/contexts/communities/requests/actions/view.cljs

```diff
@@ -79,16 +85,17 @@
     :separator [rn/view {:margin-top 1}]
     :render-fn                         community-rule-item}])
 
-(defn request-to-join [community]
-  (let [agreed-to-rules? (reagent/atom false)]
+(defn request-to-join [{:keys [permissions name id]}]
+  (let [agreed-to-rules? (reagent/atom false)
+        is-open? (not= 3 (:access permissions))]
```

Comment:
backend permissions for closed/open community. It would probably be nicer if they used a semantic naming for this instead. Can we provide feedback on that? @cammellos, @ilmotta 
wdyt?

Reply:
Great question @J-Son89. As usual, it depends, but I stand firmly in the ground that we should almost always create a well defined model of the data coming INTO the application. If the backend sends a `3`, `:three` or `1+1+1` it doesn't matter, our data model should make sure the so called `3` is transformed into something with semantic meaning as you said.

I made a comment above related to this magic number 3, I think there's another place where this number appears.

At the very least, we could extract the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) to a constant with a proper name.

We also have the `src/data_store/` directory, where various namespaces exist to implement some kind of adapter pattern or anti-corruption layer.

Reply:
Yes, this should be definitely `const` somewhere (we have many in `constants` namespace), we use enums in status-go because there is more idiomatic, but here is best to always use a reference/variable rather than the actual value

Reply:
yep, it was already there. Found it, thanks! :)